3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -27,9 +27,9 @@ exercise file.
 const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
-  // TODO Refactor this function
   const dice = [1, 2, 3, 4, 5];
-  return rollDie(1);
+  const promiseArray = dice.map((die) => rollDie(die));
```

Comment:
The `.map()` method expects a callback function here that takes a die number as its argument and return a promise. Your arrow function meets the requirement:

```js
(die) => rollDie(die)
```

However, the `rollDie` function itself also meets this requirement. So while your arrow function is fine you can also simply code this as:

```suggestion
  const promiseArray = dice.map(rollDie);
```

Reply:
I got it. thank you